common/flatpak-utils-http.c

```diff
@@ -231,6 +243,159 @@ check_http_status (guint status_code,
   return FALSE;
 }
 
+FlatpakCertificates*
+flatpak_get_certificates_for_uri(const char *uri,
+                                 GError    **error)
+{
+  g_autoptr(FlatpakCertificates) certificates = NULL;
+  g_autoptr(GUri) parsed_uri = NULL;
+  g_autofree char *hostport = NULL;
+  const char *system_certs_d = NULL;
+  g_autofree char *certs_path_str = NULL;
+  g_auto(GStrv) certs_path = NULL;
+
+  parsed_uri = g_uri_parse (uri, G_URI_FLAGS_PARSE_RELAXED, error);
+  if (!parsed_uri)
+    return NULL;
+
+  if (!g_uri_get_host (parsed_uri))
```

Comment:
Do we never canonicalize hosts? Like case, etc? (I mean, we should do whatever docker/podman do, but it makes sense to look)

Reply:
Looked at docker and I didn't find any kind of canonicalization (but I also barely understand the Go code).